en/companion_computer/holybro_pixhawk_jetson_baseboard.md

```diff
@@ -0,0 +1,1177 @@
+
+- [Setting up a Developer Environment (Toolchain)](../dev_setup/dev_env.md)
+- [Building PX4 Software](../dev_setup/building_px4.md)
+- Upload the firmware using the [command line](../dev_setup/building_px4.md#uploading-firmware-flashing-the-board) or QGroundControl (as above)
+
+### Build PX4 on Jetson
+
+You can also build and develop PX4 code on the Jetson.
+Note that there is no need to do this if you have a development computer, but you can do do so if you want.
+
+First open a terminal and SSH session into the Jetson, and enter the following command to clone the PX4 source code.
+
+```sh
+git clone https://github.com/PX4/PX4-Autopilot.git --recursive
+```
+
+::: tip
+If you don't need all branches and tags, you can get a smaller part of the repository like this:
+
+```sh
+git clone https://github.com/PX4/PX4-Autopilot.git --recursive --depth 1 --single-branch --no-tags
+```
+
+:::
+
+Then install the Ubuntu build toolchain by entering the following command
+
+```sh
+bash ./PX4-Autopilot/Tools/setup/ubuntu.sh --no-sim-tools
+```
+
+If there are warnings, you may need to run the following commands to add the path to your binary directory.
+
+```sh
+echo 'export PATH="$HOME/.local/bin:$PATH"' >> ~/.bashrc && source ~/.bashrc
+```
+
+::: tip
+If there was an error like this when running `ubuntu.sh`:
+
+```sh
+E: Unable to locate package g++-multilib
+E: Couldn't find any package by regex 'g++-multilib'
+```
+
+You will need to install the appropriate gcc toolchain as shown below, and then run `ubuntu.sh` again.
+
+```sh
+sudo apt install gcc-arm-none-eabi gdb-arm-none-eabi -y
+```
+
+:::
+
+Then give the operating system permission to use the serial ports:
+
+```sh
+sudo usermod -a -G dialout $USER
+sudo apt-get remove modemmanager -y
+```
+
+Sanity check if we can build PX4 Firmware by building PX4 for your Pixhawk hardware on the carrier board.
+Assuming you're using a Pixhawk 6x, the command would be:
+
+```sh
+make px4_fmu-v6x_default
+```
+
+If this builds successfully, you can connect the Pixhawk USB-C port to the Jetson USB port as shown below (the USB cable comes with the kit).
+
+![How to connect Jetson to Pixhawk directly](../../assets/companion_computer/holybro_pixhawk_jetson_baseboard/upload_px4_connection.png)
+
+Then upload the firmware to Pixhawk by building with the `upload` argument:
+
+```sh
+make px4_fmu-v6x_default upload
+```
+
+## Ethernet Setup using Netplan
+
+The Holybro Jetson Pixhawk carrier has an internal network switch between Pixhawk and Jetson that can be used for MAVLink or ROS 2 communication (or any other protocol).
+While you don't need to connect any external cables, you will still need to configure the Ethernet connection such that both the Jetson and PX4 are on the same subnet.
+
+PX4 has a default IP address of `10.41.10.2` (PX4 v1.15 and earlier used `192.168.0.3`), while the Jetson is configured with a default IP address of TBD.
+So you will need to TBD.
```

Comment:
@farhangnaderi Note the two TBDs here
1. What is the default setting IP/network on Jetson?
Essentially this is the same question as in https://github.com/PX4/PX4-user_guide/pull/3363/files#r1753081432 - how do we want to handle the networking - change PX4 to its previous default to match Jetson?

Reply:
Note, this section moved up because it has nothing to do with ROS2. It is just essential configuration. I am hoping you tell me that Ethernet for Jetson/Pixhawk are on same subnet by default so you wouldn't have to do anything here for PX4 v1.15.